packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/actions/ActionExecutor.java

```diff
@@ -0,0 +1,80 @@
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.view.emf.diagram.actions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.sirius.components.core.api.IFeedbackMessageService;
+import org.eclipse.sirius.components.core.api.IIdentityService;
+import org.eclipse.sirius.components.interpreter.AQLInterpreter;
+import org.eclipse.sirius.components.representations.Failure;
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.representations.Message;
+import org.eclipse.sirius.components.representations.MessageLevel;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.components.representations.WorkbenchSelection;
+import org.eclipse.sirius.components.representations.WorkbenchSelectionEntry;
+import org.eclipse.sirius.components.view.diagram.Action;
+import org.eclipse.sirius.components.view.emf.diagram.actions.api.IActionExecutor;
+import org.eclipse.sirius.components.view.emf.operations.api.IOperationExecutor;
+import org.eclipse.sirius.components.view.emf.operations.api.OperationExecutionStatus;
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to execute an action.
+ *
+ * @author arichard
+ */
+@Service
+public class ActionExecutor implements IActionExecutor {
+
+    public static final String NAMED_INSTANCES = "NAMED_INSTANCES";
+
+    private final IIdentityService identityService;
+
+    private final IOperationExecutor operationExecutor;
+
+    private final IFeedbackMessageService feedbackMessageService;
+
+    public ActionExecutor(IIdentityService identityService, IOperationExecutor operationExecutor, IFeedbackMessageService feedbackMessageService) {
+        this.identityService = Objects.requireNonNull(identityService);
+        this.operationExecutor = Objects.requireNonNull(operationExecutor);
+        this.feedbackMessageService = Objects.requireNonNull(feedbackMessageService);
+    }
+
+    @Override
+    public IStatus executeAction(Action action, AQLInterpreter interpreter, VariableManager variableManager) {
+        var result = this.operationExecutor.execute(interpreter, variableManager, action.getBody());
+
+        List<Message> feedbackStackedMessages = new ArrayList<>(this.feedbackMessageService.getFeedbackMessages());
+        if (result.status() == OperationExecutionStatus.FAILURE) {
+            List<Message> errorMessages = new ArrayList<>();
+            errorMessages.add(new Message(String.format("Something went wrong while executing the action '%s'", action.getName()), MessageLevel.ERROR));
```

Comment:
Use the message service to either reuse an existing message to to create a new one that would support internationalization

Reply:
Ok I will also update the new `ToolExecutor` where it has been done exactly in the same way.

Reply:
After trying to use `IMessageService` it seems not possible to do that here.
Indeed, `IMessageService` comes from `org.eclipse.sirius.web.domain`

Reply:
We have pretty much one message service per project, in this case, if I'm not mistaken, the one you need is `IEMFMessageService`